webroot/panel/new_group.php

```diff
@@ -0,0 +1,201 @@
+
+$types = $USER->getRequestableGroupTypes();
+$pending_requests = $USER->getPendingGroupRequests();
+
+if ($_SERVER["REQUEST_METHOD"] == "POST") {
+    $group_type_value = $_POST['group_type'];
+    $group_type_values = explode("-", $group_type_value);
+    $group_type_prefix = $group_type_values[0];
+    $group_type_slug = $group_type_values[1];
+    $group_type_time_limited = $group_type_values[2];
+    $group_type_isNameable = $group_type_values[3];
+    $group_type_exclusiveOwner = $group_type_values[4];
+
+    if ($group_type_exclusiveOwner == 1) {
+        $groups = $USER->getGroups(true);
+        $group_names = array();
+        foreach ($groups as $group) {
+            array_push($group_names, $group->getGroupUID());
+        }
+        $uid = $USER->getUID();
+        foreach ($group_names as $group_name) {
+            if (strpos($group_name, $uid) !== false) {
+                $group_type = explode("_", $group_name)[0];
+                if ($group_type == $group_type_slug) {
+                    echo "<script>alert('You already have a group of this type.');</script>";
+                    // refresh page
+                    header("Refresh:0");
+                }
+            }
+        }
+    }
+
+    if ($group_type_time_limited == 1) {
+        $group_start_date = $_POST['group_start_date'];
+        $group_end_date = $_POST['group_end_date'];
+    } else {
+        $group_start_date = null;
+        $group_end_date = null;
+    }
+
+    if ($_POST['group_name'] == "") {
+        $group_name = $USER->getUID();
+    } else {
+        $group_name = $_POST['group_name'];
+    }
+
+    $group_uid = $group_type_prefix . $group_name;
+
+    $new_group = new UnityGroup($group_uid, $LDAP, $SQL, $MAILER, $REDIS, $WEBHOOK);
+    $new_group->requestGroup(
+        $USER->getUID(),
+        $group_type_slug,
+        $group_name,
+        $SEND_PIMESG_TO_ADMINS,
+        $group_start_date,
+        $group_end_date
+    );
+    header("Refresh:0");
+}
+
+?>
+
+<h1>Request New Group</h1>
+<hr>
+
+<?php
+if (count($pending_requests) > 0) {
+    echo "<h5>Pending Requests</h5>";
+    echo "<table>";
+    foreach ($pending_requests as $request) {
+        $requested_owner = $USER;
+        echo "<tr class='pending_request'>";
+        echo "<td> 
+        <div class='type' style='border-radius: 5px; padding-left: 10px; color: white; 
+        padding-right: 10px; text-align: center; font-size: 12px; 
+        background-color: #800000'>" . $USER->getTypeNameFromSlug($types, $request['group_type']) . "</div></td>";
+        echo "<td>" . $request['group_name'] . "</td>";
+        echo "<td>" . date("jS F, Y", strtotime($request['requested_on'])) . "</td>";
+        echo "<td></td>";
+        echo "</tr>";
+    }
+    echo "</table>";
+
+    echo "<hr>";
+}
+?>
+
+<form id="newGroupForm" action="" method="POST">
+    <p>Fill in the following information to request a new group</p>
+    <div>
+        <strong>Type of Group</strong><br>
```

Comment:
For specifically the PI group type, I am unable to request a new group. The page refreshes but nothing happens. The other 2 types work fine.